prow/plank/reconciler.go

```diff
@@ -334,6 +334,11 @@ func (r *reconciler) syncPendingJob(ctx context.Context, pj *prowv1.ProwJob) (*r
 
 	pod, podExists, err := r.pod(ctx, pj)
 	if err != nil {
+		if !podExists {
```

Comment:
@NikhilSharmaWe , I don't think this implementation does what we had discussed in your previous commit, please explain why you didn't follow the ideas from there so that we can better understand the current implementation decision. 

Also please add unit test as @alvaroaleman  mentioned previously